base/gfspvgmgr/virtual_group_manager.go

```diff
@@ -84,15 +84,19 @@ func (picker *FreeStorageSizeWeightPicker) pickIndex() (uint32, error) {
 	return 0, fmt.Errorf("failed to pick weighted random index")
 }
 
-func (vgfm *virtualGroupFamilyManager) pickVirtualGroupFamily() (*vgmgr.VirtualGroupFamilyMeta, error) {
+func (vgfm *virtualGroupFamilyManager) pickVirtualGroupFamily(availableVgfIDs []uint32) (*vgmgr.VirtualGroupFamilyMeta, error) {
 	var (
 		picker   FreeStorageSizeWeightPicker
 		familyID uint32
 		err      error
 	)
 	picker.freeStorageSizeWeightMap = make(map[uint32]float64)
-	for _, f := range vgfm.vgfIDToVgf {
-		picker.addVirtualGroupFamily(f)
+	for id, f := range vgfm.vgfIDToVgf {
+		for _, vgfID := range availableVgfIDs {
```

Comment:
`for id := availableVgfIDs:
   if id in vgfIDToVgf:
       // todo
`
is more simple.

Reply:
changed